sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/state/StateDataSourceReadSuite.scala

```diff
@@ -300,35 +374,104 @@ class StateDataSourceSQLConfigSuite extends StateDataSourceTestBase {
   }
 }
 
-class HDFSBackedStateDataSourceReadSuite extends StateDataSourceReadSuite {
+class HDFSBackedStateDataSourceReadSuite
+  extends StateDataSourceReadSuite[HDFSBackedStateStoreProvider] {
```

Comment:
why adding a  HDFSBackedStateStoreProvider here? spark conf is already set below?
```
spark.conf.set(SQLConf.STATE_STORE_PROVIDER_CLASS.key,
      classOf[HDFSBackedStateStoreProvider].getName)
```

Reply:
I see the implementation below, can we load the actual statestore from spark conf here?